### PR TITLE
CWE[Quantity] and CWE[Coding] mapping inventory

### PR DESCRIPTION
**Pull request by brick-green**
This PR adds CWE[Quantity] and CWE[Coding]  from the mapping inventory.

Inventory spreadsheets:
[CWE[Coding] ](https://docs.google.com/spreadsheets/d/16EsPaP--u_j82g6YeV81ZHcJdzk0EzNbNs_hTdkek_A/edit#gid=0)
[CWE[Quantity]](https://docs.google.com/spreadsheets/d/10NSiYc6zH_no7e-kRsnq4ZcosHYAHYKSa9wPvnY-Yfk/edit#gid=0)

Test Steps:
1. run integration tests
2. compare mapping inventory sheets to implementation

## Changes
- added CWE/Coding
- added CWE/Quantity
- added integration test for CWE Quantity
- changed XON.2 mapping to use CWE[Coding]

## Linked Issues
- #11658

### CODE REVIEW

File: prime-router/metadata/fhir_mapping/hl7/datatypes/CWE/Quantity.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      valueOf: CWE.1
+      expressionType: HL7Spec
+    - condition: $text NOT_NULL
+      type: STRING
+      valueOf: CWE.2
+      expressionType: HL7Spec
+
+system:
+  condition: $identifier NOT_NULL && $nameOfCodingSystem NOT_NULL
+  type: SYSTEM_URL
```
Comment (mkalish) [inline prime-router/metadata/fhir_mapping/hl7/datatypes/CWE/Quantity.yml:29]: I'm not sure if we've done this in other places... wouldn't this mean we would only populate it if is in our URL mappings?

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XON/Organization.yml [hunk 2]
```diff
@@ -11,17 +11,23 @@ name:
...
   expressionType: nested
+  vars:
+    xon2: XON.2
   expressionsMap:
     url:
       type: SYSTEM_URL
       value: organization-name-type
     valueCodeableConcept:
-      valueOf: datatype/CodeableConcept
+      valueOf: datatypes/CWE/Coding
```
Comment (mkalish) [inline prime-router/metadata/fhir_mapping/hl7/datatypes/XON/Organization.yml:24]: I'm a little confused as to how this works.... where does the `coding` property come from?

File: prime-router/metadata/hl7_mapping/common/observation-result.yml [hunk 3]
```diff
@@ -76,11 +87,24 @@ elements:
...
 
+  - name: observation-units-identifier
+    condition: >
+      %context.value.exists() and
+      %context.value.where(extension(%`rsext-cwe-quantity`)).exists() and
+      %context.value.extension(%`rsext-cwe-quantity`).value.where(extension(%`rsext-hl7v2Name`).value = "units").exists()
+    resource: '%resource.value.extension(%`rsext-cwe-quantity`).value'
+    constants:
+      cweFieldPath: '%{hl7OBXField}-6'
+    schema: ../datatypes/codeableConcept/CWE
```
Comment (mkalish) [inline prime-router/metadata/hl7_mapping/common/observation-result.yml:106]: Should there not be a `quantity/CWE` here that takes in `%resource` and the condition checks that `value` is a `Quantity`?

**Review by mkalish — COMMENTED** [review]
Couple of thoughts on this one, let me know if you want to pair

File: prime-router/metadata/fhir_mapping/hl7/datatypes/CWE/Quantity.yml (hunk 1 above)
Comment (brick-green) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/CWE/Quantity.yml:29]: yes that's true. We do this in the current/old Coding mapping and in the new CodeableConcept mapping. But I don't rely on it for the FHIR --> HL7 mapping.  I added "L" for local codes to the files in the CodeableConcept PR.  we do this in

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XON/Organization.yml (hunk 2 above)
Comment (brick-green) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XON/Organization.yml:24]: `coding` comes from `CWE/CodeableConcept` which is referenced as an extension on `CWE/Coding`

File: prime-router/metadata/hl7_mapping/common/observation-result.yml (hunk 3 above)
Comment (brick-green) [reply prime-router/metadata/hl7_mapping/common/observation-result.yml:106]: The condition check for `value.is(Quantity)` feels like validation so I didn't want to stop a value from being populated based on that condition. I did add it to the OBX-2 mapping in this file to coincide with the existing OBX-2 mapping for "NM" but I could be convinced to leave it or get rid of it

Comment (mkalish) [reply prime-router/metadata/hl7_mapping/common/observation-result.yml:106]: so I think that check literally just confirms that the `value` is a `valueQuantity` in the FHIR.  I don't feel super strongly but your advocating for allowing someone to add the `cwe-quantity` extension to a `valueString`?

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XON/Organization.yml (hunk 2 above)
Comment (mkalish) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XON/Organization.yml:24]: hmmm.. I see, should maybe the extension just get updated to be a `valueCoding`?

Comment (brick-green) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XON/Organization.yml:24]: yeah, good idea. Made the change

File: prime-router/metadata/fhir_mapping/hl7/datatypes/CWE/Quantity.yml (hunk 1 above)
Comment (mkalish) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/CWE/Quantity.yml:28]: That makes sense to me, I'll put on the list of things we should think about for FHIR receivers

**Review by mkalish — APPROVED** [review]
:shipit:

**Pull request merged**